bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicVoice.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.voice.mimic.internal;
+
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.voice.Voice;
+
+/**
+ * Mimic Voice representation.
+ *
+ * @author Gwendal Roulleau - Initial contribution
+ */
+@NonNullByDefault
+public class MimicVoice implements Voice {
+
+    @Nullable
+    private String speaker;
+
+    private final Locale locale;
+
+    private final String key;
+
+    private final String name;
+
+    public MimicVoice(String key, String language, String name, @Nullable String speaker) {
+        this.key = key;
+        this.locale = Locale.forLanguageTag(language.replaceAll("_", "-"));
+        this.name = name;
+        this.speaker = speaker;
+    }
+
+    /**
+     * Globally unique identifier of the voice.
+     *
+     * @return A String uniquely identifying the voice globally
+     */
+    @Override
+    public String getUID() {
+        return "mimictts:" + getTechnicalName().replaceAll("[^a-zA-Z0-9_]", "");
+    }
+
+    /**
+     * Technical name of the voice.
+     *
+     * @return A String voice technical name
+     */
+    String getTechnicalName() {
+        String speakerId = (speaker != null) ? "#" + speaker : "";
+        return (key + speakerId);
+    }
+
+    @Override
+    public String getLabel() {
+        return name + ((speaker != null) ? "#" + speaker : "");
```

Comment:
Maybe you could build a better label ? Maybe with the speaker surrounding by parenthesis ?
```
return name + ((speaker != null) ? " (" + speaker + ")" : "");
```

Reply:
What is especially expected in the label is to find the language of the voice. So please add it.
In general the format is "name (language)" but you are free to add the speaker if available.

Reply:
I removed my wrong comment about adding language in the label, not required. The console command is displaying the language of each voice but using a property of the voice, not its label.

Reply:
Goog idead, done